rapids-cmake/cpm/spdlog.cmake

```diff
@@ -100,27 +100,27 @@ function(rapids_cpm_spdlog)
 
   if(_RAPIDS_FMT_OPTION STREQUAL "BUNDLED")
     set(spdlog_fmt_option "")
-  elseif(_RAPIDS_FMT_OPTION STREQUAL "EXTERNAL_FMT" OR _RAPIDS_FMT_OPTION STREQUAL
-                                                       "EXTERNAL_FMT_HO")
+  elseif(_RAPIDS_FMT_OPTION STREQUAL "EXTERNAL_FMT")
+    set(spdlog_fmt_option "SPDLOG_FMT_EXTERNAL ON")
+    set(spdlog_fmt_target fmt::fmt)
+  elseif(_RAPIDS_FMT_OPTION STREQUAL "EXTERNAL_FMT_HO")
+    set(spdlog_fmt_option "SPDLOG_FMT_EXTERNAL_HO ON")
+    set(spdlog_fmt_target fmt::fmt-header-only)
+  elseif(_RAPIDS_FMT_OPTION STREQUAL "STD_FORMAT")
+    set(spdlog_fmt_option "SPDLOG_USE_STD_FORMAT ON")
+  else()
+    message(FATAL_ERROR "Invalid option used for FMT_OPTION, got: ${_RAPIDS_FMT_OPTION}, expected one of: 'BUNDLED', 'EXTERNAL_FMT', 'EXTERNAL_FMT_HO', 'STD_FORMAT'"
+    )
+  endif()
+
+  if(_RAPIDS_FMT_OPTION STREQUAL "EXTERNAL_FMT" OR _RAPIDS_FMT_OPTION STREQUAL "EXTERNAL_FMT_HO")
     include("${rapids-cmake-dir}/cpm/fmt.cmake")
 
     # Using `spdlog_ROOT` needs to cause any internal find calls in `spdlog-config.cmake` to first
     # search beside it before looking globally.
     list(APPEND fmt_ROOT ${spdlog_ROOT})
 
     rapids_cpm_fmt(${_RAPIDS_UNPARSED_ARGUMENTS})
-
-    set(spdlog_fmt_option "SPDLOG_${_RAPIDS_FMT_OPTION} ON")
```

Comment:
Should we consider changing the valid rapids-cmake options so that they match the built-in spdlog options, i.e. make this set command work?

Reply:
I don't want to use the spdlog option as they aren't consistent ( `SPDLOG_USE_STD_FORMAT` vs `SPDLOG_FMT_EXTERNAL[_HO]` ) and am willing to have some complexity here to make a nicer user interface.